fe/fe-core/src/main/java/com/starrocks/qe/ConnectContext.java

```diff
@@ -1269,21 +1269,30 @@ public void updateByUserProperty(UserProperty userProperty) {
     public static class ScopeGuard implements AutoCloseable {
 
         private boolean set = false;
+        private ConnectContext prev;
 
         private ScopeGuard() {
         }
 
-        public static ScopeGuard setIfNotExists(ConnectContext session) {
+        private static ScopeGuard exchange(ConnectContext session) {
```

Comment:
`bind` is a better name? I also found this bug that ConnectContext in threadLocal may not be switched even if `bindScope` is called in my test: https://github.com/StarRocks/starrocks/pull/55662/files#diff-0070e8c9c2ba6be48f27c29bdc63ce38d1ac597317a23fc379a85a8788530a91

Reply:
previous logic is `setIfNotExists`, which will not change the session if there's already an thread-local session.

here we change it to `bind/exchange`, which will change the session even if it's already existed.

Reply:
should we always change threadlocal `ConnectContext` if we `bindScope` the ConnectCotext?

Reply:
I think so, i cannot think of any exceptions.